http-server/src/main/java/org/triplea/db/dao/moderator/player/info/PlayerInfoForModeratorDao.java

```diff
@@ -0,0 +1,48 @@
+package org.triplea.db.dao.moderator.player.info;
+
+import java.util.List;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+
+/**
+ * DAO used to lookup player correlation information for moderators. Answers questions such as
+ *
+ * <ul>
+ *   <li>"how many times and was this player banned?"
+ *   <li>"which other names were used by this same IP and system-id? (presumably the same player)"
+ * </ul>
+ */
+public interface PlayerInfoForModeratorDao {
+  @SqlQuery(
+      "select distinct"
+          + "    username as name,"
+          + "    ip as ip,"
+          + "    system_id as systemId,"
+          + "    max(access_time) as accessTime"
+          + "  from access_log"
+          + "  where "
+          + "    access_time > (now() - '14 day'::interval)"
+          + "    and ("
+          + "      ip = :ip::inet"
+          + "      or system_id = :systemId"
+          + "    )"
+          + "  group by name, ip, systemId"
+          + "  order by accessTime desc")
```

Comment:
Side-Note: One day we'll have [Text Blocks](https://openjdk.java.net/jeps/378) and the world will be a better place ^^

Reply:
:+1: 
Though TBH, I'm not sure when. We need the next LTS to come out and for us to be willing to migrate to it. I do think avoiding the constants between queries and the JDBI types was an improvement; that is something I think we should do going forward and see about retrofitting.